### ISSUE TITLE
Azure Functions 2.0 Support

### ISSUE THREAD

**Issue by rhavlick** [serilog/serilog#1229]
Hello, I'm just wondering if anyone has figured out a good way to get Serilog working with Azure Functions 2.0?

In Azure Functions 2.0 they changed from a TraceWriter to an ILogger.  This seems great, but unlike a MVC .NET Core app you don't have a Startup function where you can setup Serilog.  

Does anyone know of an EASY way to do this?  (I found pretty lengthy workarounds for this, but I'm not sure that's the best way to do it)

Thanks.

**Comment by MV10** [comment]
Although it isn't exactly what you're asking for (it doesn't use the Functions ILogger), for Functions V1 I wrote a DI solution specifically for injecting a Serilog Logger as a singleton. Because of some changes in V2 (and the fact that MS is adding DI to V2), I went with a temporary fix in V2 with a service locator approach. I've stopped using Functions altogether but we had this up and running for many months.

https://mcguirev10.com/2018/04/03/service-locator-azure-functions-v2.html

The problem with adding this kind of support for Functions is that it typically needs to plug into the underlying WebJobs lifecycle, and for some reason Microsoft keeps that door firmly closed. I think they're still fantasizing that Function apps will be these little hacked-together CSX script files written in the browser instead of real mini-apps (or microservices or whatever your perferred terminology might be).
